cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -297,11 +310,13 @@ struct BooleanAnyImpl : public ScalarAggregator {
     const auto& other = checked_cast<const BooleanAnyImpl&>(src);
     this->any |= other.any;
     this->has_nulls |= other.has_nulls;
+    this->count += other.count;
     return Status::OK();
   }
 
   Status Finalize(KernelContext* ctx, Datum* out) override {
-    if (!options.skip_nulls && !this->any && this->has_nulls) {
+    if ((!options.skip_nulls && !this->any && this->has_nulls) ||
+        this->count < options.min_count) {
```

Comment:
I'm curious about this condition: if there are nulls and `options.skip_nulls` is false, this kernel can still return true?

Reply:
Yes, it looked weird to me as well, but that is how [Kleene logic](https://en.wikipedia.org/wiki/Three-valued_logic#Kleene_and_Priest_logics) works, and you can observe this in R:

```r
> any(c(NA), na.rm = FALSE)
[1] NA
> any(c(NA, TRUE), na.rm = FALSE)
[1] TRUE
> any(c(NA, FALSE), na.rm = FALSE)
[1] NA
> any(c(), na.rm = FALSE)
[1] FALSE
```